common/thrift_router.cpp

```diff
@@ -51,6 +51,8 @@ bool parseHost(const std::string& str, common::detail::Host* host,
     return false;
   }
   auto group = (tokens.size() == 3 ? tokens[2] : "");
+  auto pos = group.find("_");
+  host->az = (pos != 0 && pos != std::string::npos) ? group.substr(0, pos) : group;
```

Comment:
format is "az_pg". This I got from the shard map generated by rocksplicator for the hosts it manages. If no "_" found, I default the az to be group.

Reply:
yeah, I was more referring to if we have defined the format somewhere else in the codebase --- this is to avoid hard-coding the delimiter, and potentially reusing the parsing scheme if it's already defined somewhere. 

Right now this feels very error-prone, if the format changes, the parsing here breaks

Reply:
I tried to find it in the codebase but was unable to do so. I got the format from the shard map files generated. I agree this can be error prone. however, to save us from the different formats, I am just substituting to use the whole group.

Reply:
Looks like the format is defined in the Java code: https://github.com/pinterest/rocksplicator/blob/63c9c7d738dc868e59c552a43e2b305314092b0c/cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java#L414 where the shard_map is generated, and documented here: https://github.com/pinterest/rocksplicator/blob/63c9c7d738dc868e59c552a43e2b305314092b0c/cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java#L69

Can you add a comment here so readers know why it's parsed this way?

Ideally, we should extract the parsing scheme as a common utilization function, maybe in https://github.com/pinterest/rocksplicator/blob/63c9c7d738dc868e59c552a43e2b305314092b0c/common/availability_zone.h, but that can be a follow-up discussion.

Reply:
I added the comment speciyfing this format